pkg/operator/certrotation/client_cert_rotation_controller.go

```diff
@@ -27,15 +27,41 @@ const (
 	RunOnceContextKey = "cert-rotation-controller.openshift.io/run-once"
 )
 
+// StatusReporter knows how to report the status of cert rotation
+type StatusReporter interface {
+	Report(ctx context.Context, syncErr error) (updated bool, updateErr error)
+}
+
+var _ StatusReporter = (*StaticPodConditionStatusReporter)(nil)
+
+type StaticPodConditionStatusReporter struct {
+	// name is used in operator conditions to identify this controller, compare CertRotationDegradedConditionTypeFmt.
+	name string
+
+	// Plumbing:
+	OperatorClient v1helpers.StaticPodOperatorClient
```

Comment:
move to v1helpers among the other status updating methods, add a constructor and make both the struct attributes private

Reply:
+replace `name` with `conditionName string` and pass the whole condition name in there

Reply:
> add a constructor and make both the struct attributes private

Why?